javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/LazyType.java

```diff
@@ -37,6 +38,34 @@
 
 public class LazyType implements ResolvedType {
 
+    /**
+     * Get the concrete type of a {@link ResolvedType}.
+     *
+     * This method is used to unpack the {@link LazyType} when they are nested.
+     *
+     * {@code
+     *      LazyType typeA = new LazyType(() -> ResolvedVoidType.INSTANCE);
+     *      LazyType typeB = new LazyType(() -> typeA);
+     *      LazyType typeC = new LazyType(() -> typeB);
+     * }
+     *
+     * In this example, by calling {@code getConcreteType(typeC)} we would expected the return type to be a
+     * {@link ResolvedVoidType}
+     *
+     * @param type The resolved type to find the concrete type.
+     *
+     * @return The concrete type for the {@link LazyType}.
+     */
+    public static ResolvedType getConcreteType(ResolvedType type) {
```

Comment:
I don't understand why this method is static. It seems to me that this method should return the 'concrete type' of the current LazyType

     public ResolvedType getConcreteType() {...}